src/backend/PortabilityService.Gateway/Reliability/PortabilityServiceQoSProvider.cs

```diff
@@ -0,0 +1,114 @@
+            var _retryBaseDelay = 0.5; // options.RetryBaseDelay
+            var standardRetryPolicy = Policy
+                .Handle<HttpRequestException>()
+                .Or<TimeoutException>()
+                .Or<TimeoutRejectedException>()
+                .OrResult<HttpResponseMessage>(response => ReportsServerError(response))
+                // Number of times to retry and backoff function
+                .WaitAndRetryAsync(_retryCount, i => TimeSpan.FromSeconds(Math.Pow(2, i) * _retryBaseDelay),
+                (failureResult, waitDuration, retryCount, context) =>
+                {
+                    // Log warning if retries don't work
+                    _logger.LogWarning("Retrying ({retryCount}) after {waitDuration} seconds due to: {error}",
+                        retryCount,
+                        waitDuration.TotalSeconds,
+                        GetErrorMessage(failureResult));
+                    if (failureResult.Exception != null)
+                    {
+                        _logger.LogTrace("Full exception: {exception}", failureResult.Exception.ToString());
+                    }
+                });
+
+            // Stop trying requests that repeatedly fail
+            var circuitBreakerPolicy = Policy
+                .Handle<HttpRequestException>()
+                .Or<TimeoutException>()
+                .Or<TimeoutRejectedException>()
+                // TODO : Ocelot forces policies to be one CircuitBreakPolicy (*not* CircuitBreakerPolicy<T> which is what we get
+                //        if we handle a non-exceptional response type and one TimeoutPolicy. I should look into updating it 
+                //        to use an arbitrary PolicyWrap or at least IEnumerable<IPolicy>.
+                // .OrResult<HttpResponseMessage>(response => ReportsServerError(response))
+                .CircuitBreakerAsync(options.ExceptionsAllowedBeforeBreaking, TimeSpan.FromMilliseconds(options.DurationOfBreak),
+                (exception, duration) =>
+                {
+                    // Log warning when circuit break is opened
+                    _logger.LogWarning("Circuit breaker opened for {circuitBreakerDuration}ms due to: {error}",
+                        duration.TotalMilliseconds,
+                        $"[{exception.GetType().Name}] {exception.Message}");
+                    if (exception != null)
+                    {
+                        _logger.LogTrace("Full exception: {exception}", exception.ToString());
+                    }
+                },
+                () =>
+                {
+                    // Log informational message when the circuit breaker resets
+                    _logger.LogInformation("Circuit breaker closed");
+                },
+                () =>
+                {
+                    // Log informational message when the circuit breaker is half-opem
+                    _logger.LogInformation("Circuit half-open");
+                });
```

Comment:
nit: indent the argument to `.CircuitBreakerAsync()`

Reply:
Good catch. FIxed.